core/common/assets/js/api/extras/hash-commands.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @typedef HashCommand
+ * @property {string} method,
+ * @property {string} command
+ * @property {function( ... )} callback
+ */
+
+export default class HashCommands {
+	/**
+	 * Cannot be static since it uses callback(s) that are available only after '$e' is initialized.
+	 */
+	hashFormat = {
+		'e:run': $e.run,
+		'e:route': $e.route,
+	};
+
+	constructor() {
+		this.commands = this.get();
+	}
+
+	/**
+	 * Function get().
+	 *
+	 * Get API requests that comes from hash ( eg #e:run ).
+	 *
+	 * @param {string} hash
+	 *
+	 * @returns {Array.<HashCommand>}
+	 */
+	get( hash = location.hash ) {
+		const result = [];
+
+		if ( hash ) {
+			// Remove first '#' and split each '&'.
+			const hashList = hash.substr( 1 ).split( '&' );
+
+			hashList.forEach( ( hashItem ) => {
+				const hashParts = hashItem.split( ':' );
```

Comment:
Suggestion: Maybe regex to make sure that the hash param is actually HashCommand? and not something else that another plugin or user sets in the hash (e.g: some:code:something).

Reply:
Currently, its enough, If the method does not found in `HashCommands.hashFormat` then it will fail, so it is safe. Maybe better rename it to `CommandsList` or something else...